src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -316,6 +327,100 @@ fn suggest_restriction(
 }
 
 impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
+    /// Try to find possible types that would satisfy the bounds in the type param to give an
+    /// appropriate turbofish suggestion.
+    fn get_turbofish_suggestions(
+        &self,
+        obligation: &PredicateObligation<'tcx>,
+        data: ty::TraitPredicate<'tcx>,
+        self_ty: Ty<'tcx>,
+    ) -> Vec<String> {
+        let mut turbofish_suggestions = FxHashSet::default();
+        self.tcx.for_each_relevant_impl(data.trait_ref.def_id, self_ty, |impl_def_id| {
+            let param_env = ty::ParamEnv::empty();
+            let param_env = param_env.subst(self.tcx, data.trait_ref.substs);
```

Comment:
You can probably just use `obligation.param_env` instead of risking an ICE like this.